src/cryptonote_core/cryptonote_core.h

```diff
@@ -797,6 +797,17 @@ namespace cryptonote
       */
      bool add_deregister_vote(const loki::service_node_deregister::vote& vote, vote_verification_context &vvc);
 
+     /**
+      * @brief Prepare a registration tx using the service node keys for this
+      * daemon. This function is intended to be called without being core
+      * initialized with core::init; for use when generating txs from the shell
+      *
+      * @param vm The command line variable map.
+
+      * @return whether or not the command was able to prepare the registration.
+      */
```

Comment:
I feel the more common use case is that you run this command whilst the daemon is open? The service node will be kept running on the server, when it comes time to renew, login to their instance, run the command in the daemon and get the registration command.

Or more likely they'll send an RPC command(s)? Rather that then login and have to launch the daemon with the special flag.

Reply:
Yeah, both important options and I designed it so they can be added. https://lokiproject.monday.com/boards/98875784/pulses/111888580

Reply:
Okay, I'll note that this will change around to accommodate that eventually.

Reply:
It's already on monday